### PR TITLE
bazci: skip posting issue if branch no longer exists

### PR DESCRIPTION
**Pull request by celiala**
Once we publish a maintenance release, we delete its corresponding staging branch (one of `staging-` or `release-MM.d.p-rc`). However, we run 1000s of nightly tests against all release- branches. And these TC jobs will post GH issues with incorrect branch details, if we delete these staging branches before the TC job completes.

This PR fails skips posting if staging branch no longer exists.

Epic: None.
Release note: None.
Release justification: non-production, test-infra only change.

### CODE REVIEW

**Comment by celiala** [comment]
Questions for the team:
- is there a way to test this (other than merging then delete one of the temp branches?)
- I think I need to backport this to all branches, but wanted to confirm
Thanks!

File: pkg/cmd/release/git.go [hunk 1]
```diff
@@ -286,8 +291,16 @@ func findHealthyBuild(potentialRefs []string) (buildInfo, error) {
 	return buildInfo{}, fmt.Errorf("no ref found")
 }
 
-// listRemoteBranches retrieves a list of remote branches using a pattern, assuming the remote name is `origin`.
-func listRemoteBranches(pattern string) ([]string, error) {
+// IsStagingBranch returns true if branch follows one of the following patterns:
```
Comment (celiala) [inline pkg/cmd/release/git.go:294]: I wasn't sure of the best "home" for this method, so feel free to suggest a better file/package for this?

i.e. this isn't a "git" method, but it has knowledge about release/staging branches

**Review by rail — CHANGES_REQUESTED** [review]
I managed to build it with this diff https://gist.github.com/rail/afca76fc5587f3b96578a74006f873f8

**Review by celiala — COMMENTED** [review]
:woman_facepalming: doh, yes, I didn't try building it -- thank you for the patch! CI should pass now :)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/122254#-:-NvbV79d2xLaN5lyz6GS:b-bapl9b)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @jlinder, @rail, and @rickystewart)

___
*[`pkg/cmd/bazci/bazci.go` line 532 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/122254#gh-1566656865:-NvbVKsd0m4HLjwxB1Sw:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/4c2c813aa1c79a046d3402d1429eb73cdf203987/pkg/cmd/bazci/bazci.go#L532)):*
<details><summary><i>Previously, rail (Rail Aliiev) wrote…</i></summary><blockquote>

Can you add a check to handle `err` here? Otherwise it won't compile.
</blockquote></details>

Done.
___
*[`pkg/cmd/release/git.go` line 294 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/122254#gh-1562780303:-NvbVN2T5W23nkfq-7Le:b4c9lyr) ([raw file](https://github.com/cockroachdb/cockroach/blob/5d400d0a652ce6e231248f31262ccc1c5d551e0c/pkg/cmd/release/git.go#L294)):*
<details><summary><i>Previously, rail (Rail Aliiev) wrote…</i></summary><blockquote>

We can keep it here for now. `pkg/release` would be an option, but not sure if refactoring it now is a good idea.
</blockquote></details>

:thumbsup:

**Review by rail — APPROVED** [review]
:lgtm:

> Reviewed 1 of 4 files at r2, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/122254#-:-NvaS2ZZ8-sVkzHvIuYQ:bnfp4nl)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @jlinder, @rail, and @rickystewart)

**Review by rail — APPROVED** [review]
> Reviewed 1 of 4 files at r2, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/122254)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @jlinder and @rickystewart)

**Comment by celiala** [comment]
let's see if this simpler solution does what we want: https://github.com/cockroachdb/cockroach/pull/122506/files

**Comment by celiala** [comment]
closing in favor of https://github.com/cockroachdb/cockroach/pull/122506

**Pull request closed without merging**